cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/DataPrepBrowserStore/index.js

```diff
@@ -364,6 +383,61 @@ const bigquery = (state = defaultBigQueryValue, action = defaultAction) => {
   }
 };
 
+const spanner = (state = defaultSpannerValue, action = defaultAction) => {
+  switch (action.type) {
+    // This means the user is starting afresh. Reset everything to default and set the connectionID
+    case Actions.SET_SPANNER_CONNECTION_ID:
+      return {
+        ...defaultSpannerValue,
+        connectionId: action.payload.connectionId
+      };
+    case Actions.SET_SPANNER_CONNECTION_DETAILS:
+      return {
+        ...state,
+        info: action.payload.info,
+        error: null
+      };
+    case Actions.SET_SPANNER_LOADING:
+      return {
+        ...state,
+        loading: true
+      };
+      case Actions.SET_SPANNER_INSTANCE_LIST:
+      return {
+        ...state,
+        loading: false,
+        instanceList: action.payload.instanceList,
+        instanceId: null
+      };
+    case Actions.SET_SPANNER_DATABASE_LIST:
+      return {
+        ...state,
+        loading: false,
+        instanceList: [],
+        instanceId: action.payload.instanceId,
+        databaseList: action.payload.databaseList,
+        databaseId: null
```

Comment:
This used to be a problem in bigquery when we query for tables list we reset the databaseId and viceversa (like in the next case). Can you check if the breadcrumb in the spanner browser updates correctly during navigation?

Reply:
I think this is slightly different, here we set the `databaseId` to `null` when we get and display the database list, which makes sense I think. Also the breadcrumb uses the path params to get the connectionId/databaseId/tableId, and those never get reset.